mlflow/resource_mlflow_model.go

```diff
@@ -3,112 +3,72 @@ package mlflow
-	Status               string `json:"status,omitempty"`
-}
-
-// Model defines the response object from the API
-type Model struct {
-	Name                 string         `json:"name" tf:"force_new"`
-	CreationTimestamp    int64          `json:"creation_timestamp,omitempty" tf:"computed"`
-	LastUpdatedTimestamp int64          `json:"last_updated_timestamp,omitempty" tf:"computed"`
-	UserID               string         `json:"user_id,omitempty" tf:"computed"`
-	LatestVersions       []ModelVersion `json:"latest_versions,omitempty" tf:"computed"`
-	Description          string         `json:"description,omitempty"`
-	Tags                 []Tag          `json:"tags,omitempty"`
-	RegisteredModelID    string         `json:"id,omitempty" tf:"computed,alias:registered_model_id"`
-}
-
-// registeredModel defines response from GET API op
-type registeredModel struct {
-	RegisteredModelDatabricks Model `json:"registered_model_databricks"`
-}
-
-type ModelsAPI struct {
-	client  *common.DatabricksClient
-	context context.Context
-}
-
-func NewModelsAPI(ctx context.Context, m any) ModelsAPI {
-	return ModelsAPI{m.(*common.DatabricksClient), ctx}
-}
-
-func (a ModelsAPI) Create(m *Model) error {
-	return a.client.Post(a.context, "/mlflow/registered-models/create", m, m)
-}
-
-func (a ModelsAPI) Read(name string) (*Model, error) {
-	var m registeredModel
-	err := a.client.Get(a.context, "/mlflow/databricks/registered-models/get", map[string]string{
-		"name": name,
-	}, &m)
-	if err != nil {
-		return nil, err
-	}
-	return &m.RegisteredModelDatabricks, nil
-}
-
-// Update the model entity
-func (a ModelsAPI) Update(m *Model) error {
-	return a.client.Patch(a.context, "/mlflow/registered-models/update", m)
-}
-
-// Delete removes the model by its name
-func (a ModelsAPI) Delete(name string) error {
-	return a.client.Delete(a.context, "/mlflow/registered-models/delete", map[string]string{
-		"name": name,
-	})
-}
-
 func ResourceMlflowModel() *schema.Resource {
 	s := common.StructToSchema(
-		Model{},
-		func(m map[string]*schema.Schema) map[string]*schema.Schema {
-			delete(m, "latest_versions")
-			return m
+		ml.Model{},
+		func(s map[string]*schema.Schema) map[string]*schema.Schema {
+			delete(s, "latest_versions")
+			s["name"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			}
+			return s
 		})
 
 	return common.Resource{
 		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			var m Model
-			common.DataToStructPointer(d, s, &m)
-			if err := NewModelsAPI(ctx, c).Create(&m); err != nil {
+			w, err := c.WorkspaceClient()
+			if err != nil {
 				return err
 			}
-			d.SetId(m.Name)
+			var req ml.CreateModelRequest
+			common.DataToStructPointer(d, s, &req)
+			res, err := w.ModelRegistry.CreateModel(ctx, req)
+			if err != nil {
+				return err
+			}
+			d.SetId(res.RegisteredModel.Name)
+			d.Set("registered_model_id", res.RegisteredModel.Name) // alias
```

Comment:
didn't we agree it's supposed to be `res.RegisteredModel.Id`? 

see: 

> 	RegisteredModelID string `json:"id,omitempty" tf:"computed,alias:registered_model_id"`

Reply:
> didn't we agree it's supposed to be `res.RegisteredModel.Id`?
> 
> see:
> 
> > RegisteredModelID string `json:"id,omitempty" tf:"computed,alias:registered_model_id"`

yeah but `w.ModelRegistry.CreateModel` returns an [ml.Model](https://github.com/databricks/databricks-sdk-go/blob/main/service/ml/model.go#L800) which doesn't have an ID field, which changed in the [most recent PR](https://github.com/databricks/databricks-sdk-go/pull/370/files#). Also, model name and d.ID should be identical (and are based on the prior line).

Reply:
@mberk06 make an additional call to get model